### DISCUSSION TITLE
can't prevent scripts from loading when they generated with append_javascript_pack_tag

### DISCUSSION THREAD

**Discussion by benjamindickens** [shakacode/shakapacker#442] [Q&A]
## Desired behavior:

i want method "append_javascript_pack_tag" to be able to change src attribute to data-src in script tag or add another method to get list with urls of all generated chunks and be able to pass as string to some custome data-attr.
## Actual behavior:
right know i really need to delay loading of all scripts(not executing) on my page and i am not able to. i have been searching a method to help me parse this data some how from this method append_javascript_pack_tag to get src of each generated script and add it myself with some inline js after some event; please help me somehow (( may be there is some another way to complete the task 

## Setup environment:
- Ruby version: ruby-3.2.0
- Rails version: 
- Shakapacker version:

**Comment by tomdracz** [comment]
Converted it from issue info discussion as that feels more sensible.

`append_javascript_pack_tag` only adds files to render queue, it's the `javascript_pack_tag` declaration that loads them in as script tags.

Your best bet is probably monkey patching the methods. I think `javascript_pack_tag_queue` https://github.com/shakacode/shakapacker/blob/49e4b2c4dfe2c8e0999c3cde2ac4dcfe7df14f01/lib/shakapacker/helper.rb#L105 would give you the list of sources that you can then pass to any custom handling
